pkg/plugin/test-keeper/plugin/test_checker.go

```diff
@@ -12,23 +12,48 @@ type TestChecker struct {
 	TestKeeperMatcher TestMatcher
 }
 
-// IsAnyNotExcludedFileTest checks if a commit affects any test file
-func (checker *TestChecker) IsAnyNotExcludedFileTest(files []scm.ChangedFile) (bool, error) {
+// FileCategories holds information about the total files coming in the changeset, legit files (those which are excluded from test verification)
+// and Tests
+type FileCategories struct {
+	Total, Legit, Tests int
+	Files               *[]scm.ChangedFile
+}
+
+// OnlyLegitFiles indicates if changeset contains only files which are excluded from test verification
+func (f *FileCategories) OnlyLegitFiles() bool {
+	return f.Total > 0 && f.Legit == f.Total
```

Comment:
do we really need to check the total count?
btw, if there is a PR without any file different, then it is automatically merged/closed

Reply:
There won't be 0 files in GitHub PR, that is true. But for the boundary case, if the changeset has 0 files than it have no tests and no skipped files - so answering to "onlySkippedFiles" should be false, right?

Reply:
Agree. 
Last question  - do we need to also check the boundary case when there is no changed file in the PR - in the event_handler? Using common sense I would say no. As a pedantic guy probably yes :-D

Reply:
That's unrealistic assuming it is coming from GitHub... So I would say YAGNI? ;)